cns/simulator/simulator.go

```diff
@@ -65,93 +56,153 @@ type CnsVolumeManager struct {
 const simulatorDiskUUID = "6000c298595bf4575739e9105b2c0c2d"
 
 func (m *CnsVolumeManager) CnsCreateVolume(ctx *simulator.Context, req *cnstypes.CnsCreateVolume) soap.HasFault {
+	vctx := ctx.For(vim25.Path)
+	vsom := vctx.Map.VStorageObjectManager()
+
 	task := simulator.CreateTask(m, "CnsCreateVolume", func(*simulator.Task) (vim25types.AnyType, vim25types.BaseMethodFault) {
 		if len(req.CreateSpecs) == 0 {
 			return nil, &vim25types.InvalidArgument{InvalidProperty: "CnsVolumeCreateSpec"}
 		}
 
-		operationResult := []cnstypes.BaseCnsVolumeOperationResult{}
+		var operationResult []cnstypes.BaseCnsVolumeOperationResult
+
 		for _, createSpec := range req.CreateSpecs {
-			staticProvisionedSpec, ok := interface{}(createSpec.BackingObjectDetails).(*cnstypes.CnsBlockBackingDetails)
-			if ok && staticProvisionedSpec.BackingDiskId != "" {
-				datastore := simulator.Map.Any("Datastore").(*simulator.Datastore)
+			details, ok := createSpec.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails)
+			if ok && details.BackingDiskId != "" {
```

Comment:
If you invert the logic here so `if !ok || details.BackingDiskId == "" { continue; }` then you could dedent the large block below.